evm/src/keccak_sponge/columns.rs

```diff
@@ -72,12 +73,35 @@ pub(crate) struct KeccakSpongeColumnsView<T: Copy> {
     /// The first part of the state of the sponge, seen as bytes, after the permutation is applied.
     /// This also represents the output digest of the Keccak sponge during the squeezing phase.
     pub updated_digest_state_bytes: [T; KECCAK_DIGEST_BYTES],
+
+    /// The counter column (used for the range check) starts from 0 and increments.
+    pub range_counter: T,
+    /// The frequencies column used in logUp.
+    pub rc_frequencies: T,
 }
 
 // `u8` is guaranteed to have a `size_of` of 1.
 /// Number of columns in `KeccakSpongeStark`.
 pub const NUM_KECCAK_SPONGE_COLUMNS: usize = size_of::<KeccakSpongeColumnsView<u8>>();
 
+// Indices for LogUp range-check.
+// They are on the last registers of this table.
+pub(crate) const RC_FREQUENCIES: usize = NUM_KECCAK_SPONGE_COLUMNS - 1;
+pub(crate) const RANGE_COUNTER: usize = RC_FREQUENCIES - 1;
+
+/// Indices for the range-checked values, i.e. the `block_bytes` section.
+// TODO: Find a better way to access those indices
+pub(crate) const fn get_block_bytes_range() -> Range<usize> {
+    let start = 6 + KECCAK_RATE_BYTES + KECCAK_RATE_U32S + KECCAK_CAPACITY_U32S;
+    start..start + KECCAK_RATE_BYTES
+}
```

Comment:
I'm not super happy with this, as it can be broken easily if forgotten during a trace update.
One way to handle it properly may be to use an Enum for the different components inside the trace, and implement `Index` on the `KeccakSpongeColumnsView` with it?

Reply:
I don't know enough to have much to say here, but if we're talking about a set of sections with varying sizes I think a custom `Index` impl would be a bit odd. If the sections were all of the same size, then perhaps yes.

How about this:

```rust
use std::ops::Range;
pub enum Sections {
    BlockBytes,
    Other,
    Moar,
}

impl Sections {
    const fn range(&self) -> Range<usize> {
        match self {
            // This is a bit awkwardly long, but it's `const`
            Self::BlockBytes => {
                (6 + KECCAK_RATE_BYTES + KECCAK_RATE_U32S + KECCAK_CAPACITY_U32S)
                    ..(6 + KECCAK_RATE_BYTES
                        + KECCAK_RATE_U32S
                        + KECCAK_CAPACITY_U32S
                        + KECCAK_RATE_BYTES)
            }
            Self::Other => 1..11,
            Self::Moar => 129..890,
        }
    }
}

pub(crate) const fn single_block_bytes_val(i: usize) -> usize {
    debug_assert!(i < KECCAK_RATE_BYTES);
    // Using  it is nice
    Sections::BlockBytes.range().start + i
}
```

Another idea is to use a module and constants:

```rust
#[allow(non_snake_case, non_upper_case_globals)]
pub mod Sections {
     pub const BlockBytes: Range<usize> = 123..456,
     pub const Other:  Range<usize> = 789..1000,
     pub const Moar:  Range<usize> = 9..10
}
```

Reply:
I don't think this solves the problem in the end. Because altering the ordering / registers somehow will risk to mess this up. As we need the constraints for the milestone, I'll merge as is following @LindaGuiga's approval, but I'll come back to this topic, as a clearer way to handle it here (and in the other files) would be beneficial.